ta/pkcs11/src/pkcs11_token.c

```diff
@@ -900,6 +900,17 @@ enum pkcs11_rc entry_ck_token_initialize(uint32_t ptypes, TEE_Param *params)
 
 	update_persistent_db(token);
 
+	/* Remove all persistent objects */
+	if (token->db_objs && token->db_objs->count > 0) {
+		struct pkcs11_object *obj = NULL;
+
+		while (!LIST_EMPTY(&token->object_list)) {
+			obj = LIST_FIRST(&token->object_list);
+			unregister_persistent_object(token, obj->uuid);
```

Comment:
Could be due to a (temporary) shortage of available memory. I agree that triggering a panic instead of leaving the object in place is preferable. I'll include your suggestion. Thanks!